libraries/AP_Camera/AP_RunCam.cpp

```diff
@@ -244,7 +244,7 @@ bool AP_RunCam::pre_arm_check(char *failure_msg, const uint8_t failure_msg_len)
     }
 
     // currently in the OSD menu, do not allow arming
-    if (is_arming_prevented()) {
```

Comment:
please don't change these - it makes it less clear. I don't think this is what @peterbarker was proposing, he suggested having two functions and I think that would be fine.